### PR TITLE
Traget Panning and some ui touches

### PR DESCRIPTION
**Pull request by cquiroz**
This PR makes the explore target not change when we do panning in aladin. Instead there is an internal target that we track and visualizations are drawn referring to the internal target.
FOV and current cursor position is also displayed now outside aladin and a center target button is also included

### CODE REVIEW

File: targeteditor/src/main/scala/explore/targeteditor/AladinContainer.scala [hunk 1]
```diff
@@ -37,11 +54,54 @@ final case class AladinContainer(
...
+    * On the state we keep the svg to avoid recalculations during panning
+    */
+  @Lenses
+  final case class State(svg: Option[Svg], fov: Fov, current: Coordinates)
+
+  object State {
+    val Zero: State = State(None, Fov(Angle.Angle0, Angle.Angle0), Coordinates.Zero)
+  }
+
+  // TODO: We may want to move these to gsp-math
```
Comment (rpiaggio) [inline targeteditor/src/main/scala/explore/targeteditor/AladinContainer.scala:67]: We should also probably unify with how it's shown in the elevation plots. Right now, in the plots I'm showing with the degrees, seconds and minutes symbols. What's the usual format for angles?

File: targeteditor/src/main/scala/explore/targeteditor/AladinContainer.scala [hunk 2]
```diff
@@ -95,68 +155,194 @@ object AladinContainer {
...
+        CallbackTo
+          .pure(
+            visualization
+              .shapesToSvg(GmosGeometry.shapes(p.options.posAngle),
+                           GmosGeometry.pp,
+                           v.pixelScale,
+                           GmosGeometry.ScaleFactor
+              )
+          )
+          .flatTap(svg => $.modState((s: State) => s.copy(svg = svg.some, fov = v.fov)))
```
Comment (rpiaggio) [inline targeteditor/src/main/scala/explore/targeteditor/AladinContainer.scala:185]: Is the type ascription required?

File: targeteditor/src/main/scala/explore/targeteditor/AladinContainer.scala [hunk 3]
```diff
@@ -37,11 +54,54 @@ final case class AladinContainer(
 object AladinContainer {
   type Props = AladinContainer
 
+  /**
+    * On the state we keep the svg to avoid recalculations during panning
```
Comment (rpiaggio) [inline targeteditor/src/main/scala/explore/targeteditor/AladinContainer.scala:58]: Nice

File: targeteditor/src/main/scala/explore/targeteditor/AladinContainer.scala (hunk 1 above)
Comment (cquiroz) [reply targeteditor/src/main/scala/explore/targeteditor/AladinContainer.scala:67]: In this case they are not really angles, they are coordinates so using : seems right
FOV is special as we want to show in degrees if it is a large FOV while going down to mins/seconds/etc when they get smaller

File: targeteditor/src/main/scala/explore/targeteditor/AladinContainer.scala (hunk 2 above)
Comment (cquiroz) [reply targeteditor/src/main/scala/explore/targeteditor/AladinContainer.scala:185]: Yeah, it didn't compile for me otherwise

**Pull request merged**